packages/legend-query/src/stores/QueryBuilderPropertyEditorState.ts

```diff
@@ -115,6 +117,68 @@ export const getPropertyPath = (
   return propertyNameChain.join('.');
 };
 
+export const generateValueSpecificationForParameter = (
+  parameter: VariableExpression,
+  queryBuilderState: QueryBuilderState,
```

Comment:
Let's pass in the `graph` here instead, this function has the potential to be more generic and moved out as some common utility.